source/common/upstream/subset_lb.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "envoy/common/optional.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/upstream/load_balancer.h"
+
+#include "common/common/macros.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+#include "common/upstream/upstream_impl.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class SubsetLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::upstream> {
+public:
+  SubsetLoadBalancer(LoadBalancerType lb_type, HostSet& host_set, const HostSet* local_host_set,
+                     ClusterStats& stats, Runtime::Loader& runtime,
+                     Runtime::RandomGenerator& random, const LoadBalancerSubsetInfo& subsets);
+  ~SubsetLoadBalancer();
+
+  // Upstream::LoadBalancer
+  HostConstSharedPtr chooseHost(LoadBalancerContext* context) override;
+
+private:
+  const LoadBalancerType lb_type_;
```

Comment:
nit: Can we put all variables together. In the rest of the code we do in each visibility section:
1) struct/inner class definitions
2) method definitions
3) variables